src/state.ts

```diff
@@ -0,0 +1,24 @@
+import {Event} from './event';
+
+export class State {
+	public readonly events;
+
+	constructor() {
+		this.events = {};
+	}
+
+	async save(event: Event): Promise<unknown> {
+		if (event.key && event.value) {
+			await this.events.push(event);
+		}
+		return this.events;
+	}
+
+	async retrieve(key): Promise<unknown> {
+		for (const prop in this.events) {
```

Comment:
Keys in this system should be unique meaning we shouldn’t need to iterate over them. a key lookup or retrieval should be an O(n) operation here since we know the key ame ahead of time and either:
1) you find the key and return its value or
2) you don’t find the key and return null